GitUI/SpellChecker/WordAtCursorExtractor.cs

```diff
@@ -1,41 +1,83 @@
-﻿using System.Text;
+﻿using System;
+using System.Text;
 
 namespace GitUI.SpellChecker
 {
     internal class WordAtCursorExtractor : IWordAtCursorExtractor
     {
         public string Extract(string text, int cursor)
         {
+            var startIndex = FindStartOfWord(text, cursor);
+            return startIndex < 0 ? string.Empty : text.Substring(startIndex, cursor - startIndex + 1);
+        }
+
+        public (int start, int length) FindWord(string text, int cursor)
+        {
+            int start = Math.Min(FindStartOfWord(text, cursor), cursor);
+            if (start < 0)
+            {
+                return (-1, 0);
+            }
+
+            int end = FindEndOfWord(text, cursor);
+            return (start, Math.Max(1, end - start));
+        }
+
+        public int FindStartOfWord(string text, int cursor)
+        {
+            cursor = Math.Min(cursor, text.Length - 1);
             if (cursor < 0)
             {
-                return string.Empty;
+                return -1;
             }
 
-            var sb = new StringBuilder();
+            // do not avoid that the result can be right from the initial cursor position
+            #if false
+            if (!BelongsToStartOfWord(text, cursor))
+            {
+                return cursor;
+            }
+            #endif
```

Comment:
Is this required?

Reply:
No, it would be required for an intuitive implemention only. Though the function `Extract()`, whose behviour shall not change, has some special requirements - see the test case `"Introduce .babeljs]", ExpectedResult = ""` which shall not return the trailing "]".
I did not remove the deactivated code and added a comment to the `#if false` lest the next one should think this function was not implemented correctly.